src/core/services/aggregation.js

```diff
@@ -106,28 +129,63 @@ export default class Aggregation {
 		return sum;
 	}
 
-	static join(rows, getValue) {
+	static join(rows, getValue, aggregationOptions) {
 		if (!rows.length) {
 			return null;
 		}
 
-		let length = rows.length,
+		let separator = aggregationOptions.separator || '',
+			length = rows.length,
 			i = 0,
 			join = getValue(rows[i++]);
 
+		if (aggregationOptions.distinct) {
+			let uniqueValues = new Set(),
```

Comment:
Yes, we can. But it'll works slowly, because on calculating sum, or getting join, we will walk through array twice.